compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -1973,6 +1992,13 @@ private BType checkIndexAccessExpr(BLangIndexBasedAccess indexBasedAccessExpr, B
                 checkExpr(indexExpr, this.env);
                 actualType = symTable.xmlType;
                 break;
+            case TypeTags.TUPLE:
+                indexExprType = checkIndexExprForTupleFieldAccess(indexExpr);
+                if (indexExprType.tag == TypeTags.INT) {
```

Comment:
what should happen for other types? (example float?)

Reply:
Float validation happens in checkIndexExprForTupleFieldAccess method it seems.